File: src/flapping_cage.py
import requests

class FlappingCage:
	def __init__(self) -> None:
		self.api = "http://www.flappingcage.com/FC"
		self.headers = {
			"user-agent": "libcurl-agent/1.0"
		}

	def rank_db(
			self,
			statistics: list,
			values: list) -> str:
		data = dict(zip(statistics, values))
		return requests.post(
			f"{self.api}/rankdb.php",
			data=data,
			headers=self.headers).text

	def insert_score(
			self,
			statistics: list,
			values: list) -> str:
		data = dict(zip(statistics, values))
		return requests.post(
			f"{self.api}/ins_sc.php",
			data=data,
			headers=self.headers).text

	def get_yest_win(self) -> str:
		return requests.get(
			f"{self.api}/yest_win.php",
			headers=self.headers).text

File: src/flapping_cage.js
class FlappingCage {
	constructor() {
		this.api = "http://www.flappingcage.com/FC"
		this.headers = {
			"user-agent": "libcurl-agent/1.0"
		}
	}

	async rankDb(statistics, values) {
		const body = Object.fromEntries(zip(statistics, values))
		const response = await fetch(
			`${this.api}/rankdb.php`, {
				method: "POST",
				body: JSON.stringify(body),
				headers: this.headers
			})
		return response.text()
	}

	async insertScore(statistics, values) {
		const body = Object.fromEntries(zip(statistics, values))
		const response = await fetch(
			`${this.api}/ins_sc.php`, {
				method: "POST",
				body: JSON.stringify(body),
				headers: this.headers
			})
		return response.text()
	}

	async getYestWin() {
		const response = await fetch(
			`${this.api}/yest_win.php`, {
				method: "GET",
				headers: this.headers
			})
		return response.text()
	}
}

module.exports = {FlappingCage}
